frontend/src/pages/projects/screens/detail/data-connections/ManageDataConnectionModal.tsx

```diff
@@ -53,27 +72,62 @@ const ManageDataConnectionModal: React.FC<ManageDataConnectionModalProps> = ({
       existingData?.data.metadata.name,
     );
 
-    const handleError = (e: Error) => {
-      setErrorMessage(e.message || 'An unknown error occurred');
-      setIsProgress(false);
-    };
+    const promiseActions: Promise<K8sResourceCommon>[] = [];
+    const secretName = assembledSecret.metadata.name;
 
     if (existingData) {
-      replaceSecret(assembledSecret)
-        .then(() => onBeforeClose(true))
-        .catch(handleError);
+      if (AWSDataChanged) {
+        promiseActions.push(replaceSecret(assembledSecret));
+      }
+
+      const notebooksToDisconnect = allAvailableNotebooks.filter((notebook) =>
+        removedConnections.includes(notebook.metadata.name),
+      );
+      notebooksToDisconnect.forEach((notebook) => {
+        const secretEnvFroms = getSecretsFromList(notebook);
+        const newSecretEnvFroms = secretEnvFroms.filter(
+          ({ secretRef: { name } }) => name !== secretName,
+        );
+        if (secretEnvFroms.length === newSecretEnvFroms.length) {
+          // nothing changed, ignore -- shouldn't happen but no sense in patching
+          return;
+        }
+        promiseActions.push(
+          replaceNotebookSecret(notebook.metadata.name, projectName, newSecretEnvFroms),
+        );
+      });
     } else {
-      createSecret(assembledSecret)
-        .then(() => {
-          onBeforeClose(true);
-        })
-        .catch(handleError);
+      promiseActions.push(createSecret(assembledSecret));
+    }
+
+    const notebooksToConnect = allAvailableNotebooks.filter((notebook) =>
+      addedConnections.includes(notebook.metadata.name),
+    );
+
+    notebooksToConnect.forEach((notebook) =>
```

Comment:
Same comment here too... I just find `.forEach` usages should not push into an array, they should be used for side-effects.